tests/unit/expire.tcl

```diff
@@ -261,100 +275,286 @@ start_server {tags {"expire"}} {
-        r set foo9 bar
-        r getex foo9 EX 100
-        r set foo10 bar
-        r getex foo10 PX 100000
-        r set foo11 bar
-        r getex foo11 EXAT [expr [clock seconds] + 100]
-        r set foo12 bar
-        r getex foo12 PXAT [expr [clock milliseconds] + 100000]
-
-        after 2000
-        r debug loadaof
-        assert_range [r ttl foo1] 90 98
-        assert_range [r ttl foo2] 90 98
-        assert_range [r ttl foo3] 90 98
-        assert_range [r ttl foo4] 90 98
-        assert_range [r ttl foo5] 90 98
-        assert_range [r ttl foo6] 90 98
-        assert_range [r ttl foo7] 90 98
-        assert_range [r ttl foo8] 90 98
-        assert_range [r ttl foo9] 90 98
-        assert_range [r ttl foo10] 90 98
-        assert_range [r ttl foo11] 90 98
-        assert_range [r ttl foo12] 90 98
+    # Start a new server with empty data and AOF file.
+    start_server {overrides {appendonly {yes} appendfilename {appendonly.aof} appendfsync always}} {
+        test {All time-to-live(TTL) in commands are propagated as absolute timestamp in milliseconds in AOF} {
+            # This test makes sure that expire times are propagated as absolute
+            # times to the AOF file and not as relative time, so that when the AOF
+            # is reloaded the TTLs are not being shifted forward to the future.
+            # We want the time to logically pass when the server is restarted!
+
+            set aof [file join [lindex [r config get dir] 1] [lindex [r config get appendfilename] 1]]
+
+            # Apply each TTL-related command to a unique key
+            # SET commands
+            r set foo1 bar ex 100
+            r set foo2 bar px 100000
+            r set foo3 bar exat [expr [clock seconds]+100]
+            r set foo4 bar pxat [expr [clock milliseconds]+100000]
+            r setex foo5 100 bar
+            r psetex foo6 100000 bar
+            # EXPIRE-family commands
+            r set foo7 bar
+            r expire foo7 100
+            r set foo8 bar
+            r pexpire foo8 100000
+            r set foo9 bar
+            r expireat foo9 [expr [clock seconds]+100]
+            r set foo10 bar
+            r pexpireat foo10 [expr [clock seconds]*1000+100000]
+            r set foo11 bar
+            r expireat foo11 [expr [clock seconds]-100]
+            # GETEX commands
+            r set foo12 bar
+            r getex foo12 ex 100
+            r set foo13 bar
+            r getex foo13 px 100000
+            r set foo14 bar
+            r getex foo14 exat [expr [clock seconds]+100]
+            r set foo15 bar
+            r getex foo15 pxat [expr [clock milliseconds]+100000]
+            # RESTORE commands
+            r set foo16 bar
+            set encoded [r dump foo16]
+            r restore foo17 100000 $encoded
+            r restore foo18 [expr [clock milliseconds]+100000] $encoded absttl
+
+            # Sleep some to allow AOF to be flushed.
+            after 2000
```

Comment:
this sleep was in order to let the time pass (not in order to flush the AOF file). 
in fact, as soon as you got the reply for the last command you executed, this contents is already in the AOF file (`appendfsync always` isn't needed, it's about power-loss and kernel panics)

i suggest to drop the comment, and move that sleep to be just before that `debug loadaof` below.

Reply:
Thanks for the info. I didn't realize that. Changed.

I added a comment to say this sleep is to "let time pass", to warn other uninformed people like me.